src/knot_vector.h

```diff
@@ -0,0 +1,57 @@
+/* Copyright 2018 Chair for Computational Analysis of Technical Systems, RWTH Aachen University
+
+This file is part of SplineLib.
+
+SplineLib is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation version 3 of the License.
+
+SplineLib is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with SplineLib.  If not, see
+<http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SPLINELIB_KNOTVECTOR_H
+#define SPLINELIB_KNOTVECTOR_H
+
+#include <initializer_list>
+#include <limits>
+#include <stdexcept>
+#include <utility>
+#include <vector>
+
+#include <iostream>
+typedef std::vector<double>::const_iterator ConstKnotIterator;
+
+class KnotVector {
+ public:
+  KnotVector() = default;
+  explicit KnotVector(std::vector<double> knots);
+  KnotVector(std::initializer_list<double> knots);
```

Comment:
This constructor should have `explicit` keyword.

Reply:
This leads to the error message:
SplineLib/test/b_spline_test.cc(25): error: copy-list-initialization cannot use a constructor marked "explicit"
      KnotVector knot_vector = {0, 0, 0, 1, 2, 3, 4, 4, 5, 5, 5};